sources/stat.h

```diff
@@ -65,8 +70,9 @@ od_stat_query_end(od_stat_t *stat,
 			*query_time = diff;
 			od_atomic_u64_add(&stat->query_time, diff);
 			od_atomic_u64_inc(&stat->count_query);
-			if (stat->query_hgram)
-				od_hgram_add_data_point(stat->query_hgram, diff);
+			if (stat->enable_quantiles) {
+				td_add(stat->query_hgram[stat->current_tdigest], diff, 1);
```

Comment:
Is it thread safe? Will it work with concurrent merges?

Reply:
It isn't thread-safe. I fix it with the same trick as you - using memcpy.